ansible/roles/devspace/tasks/main.yml

```diff
@@ -0,0 +1,159 @@
+      name: "{{ item }}"
+      #extra_args: "--upgrade"
+      state: latest
+  with_items:
+    - pip
+    - passlib
+    - pyyaml
+    - docker-compose
+
+# TEMP: Error: docker-py version is 1.10.2. Minimum version required is 1.7.0.
+# see https://github.com/ansible/ansible/issues/17495
+- name: TODO install docker-py
+  become: yes
+  pip:
+      name: "{{ item }}"
+      state: latest
+  with_items:
+    - docker-py==1.9.0
+
+- name: clone devspace
+  become: yes
+  become_user: "{{ devuser }}"
+  git:
+      repo="{{ git_repo }}"
+      dest={{ devhome }}
+      update=no
+      refspec=+refs/pull/*:refs/heads/*
+      version={{ version }}
+
+- name: ensures .ssh dir exists
+  become: yes
+  become_user: "{{ devuser }}"
+  file:
+      path={{ devhome }}/slave/.ssh
+      state=directory
+      owner={{ devuser }}
+      group={{ devuser }}
+      mode=0700
+
+- name: copy snoopy
+  become: yes
+  become_user: "{{ devuser }}"
+  copy:
+       src={{ snoopy_dir_path }}/{{ item.srcfile }}
+       dest={{ devhome }}/slave/{{ item.dest }}
+       owner={{ devuser }}
+       group={{ devuser }}
+       mode=0700
+  with_items:
+    - { srcfile: ".ssh/config", dest: ".ssh/config" }
+    - { srcfile: ".ssh/snoopycrimecop_github", dest: ".ssh/snoopycrimecop_github" }
+    - { srcfile: ".ssh/snoopycrimecop_github.pub", dest: ".ssh/snoopycrimecop_github.pub" }
+    - { srcfile: ".gitconfig", dest: ".gitconfig" }
+  when: snoopy_dir_path|length>0
+
+- name: known_hosts
+  become: yes
+  become_user: "{{ devuser }}"
+  shell: ssh-keyscan github.com >> {{ devhome }}/slave/.ssh/known_hosts
+- name: chmod known_hosts
+  become: yes
+  become_user: "{{ devuser }}"
+  file:
+      path={{ devhome }}/slave/.ssh/known_hosts
+      mode=0700
+
+- name: change USER_ID
+  become: yes
+  become_user: "{{ devuser }}"
+  replace:
+      dest="{{ devhome }}/{{ item.file }}"
+      regexp='^ARG USER_ID=1000'
+      replace='ARG USER_ID={{ user_id }}' # how to find out UID?
+  with_items:
+    - { file: nginx/Dockerfile }
+    - { file: slave/Dockerfile }
+    - { file: server/Dockerfile }
+    - { file: web/Dockerfile }
+
+- name: set basic auth password for jenkins
+  become: yes
+  lineinfile:
+    dest: "{{ devhome }}/jenkins/conf.d/jenkins.conf"
+    backup: yes
+    line: "        auth_basic \"Restricted\";\n        auth_basic_user_file /etc/nginx/conf.d/passwdfile;"
+    insertafter: "    location / {"
+    state: present
+  when: jenkins_nginx_username|length>0 and jenkins_nginx_password|length>0
+
+- name: set jenkins password
+  become: yes
+  become_user: "{{ devuser }}"
+  htpasswd:
+    path="{{ devhome }}/jenkins/conf.d/passwdfile"
+    name={{ jenkins_nginx_username }}
+    password={{ jenkins_nginx_password }}
+  when: jenkins_nginx_username|length>0 and jenkins_nginx_password|length>0
+
+- name: ssl cert for jenkins
+  become: yes
+  become_user: "{{ devuser }}"
+  shell: "{{ devhome }}/sslcert {{ devhome }}/jenkins/sslcert {{ openstack_ip }}"
```

Comment:
Also is it possible to use `creates:` so it only runs if the cert hasn't already been created?